api.go

```diff
@@ -8,26 +8,26 @@ import (
 
 	"github.com/maniksurtani/quotaservice/config"
 	"github.com/maniksurtani/quotaservice/logging"
-
-	pbconfig "github.com/maniksurtani/quotaservice/protos/config"
+	pb "github.com/maniksurtani/quotaservice/protos/config"
 )
 
 // The Server interface is what you get when you create a new quotaservice.
 type Server interface {
 	Start() (bool, error)
 	Stop() (bool, error)
 	SetLogger(logger logging.Logger)
-	ServeAdminConsole(*http.ServeMux, string, config.ConfigPersister, bool)
+	CreateBucketContainer(*pb.ServiceConfig)
```

Comment:
Yeah, there are a lot of discussions on optional params for Go. Most of the strategies are discussed [here](http://joneisen.tumblr.com/post/53695478114/golang-and-default-values).

I'd still favor passing it in the factory method, allowing a `nil` value (and documenting so) which would cause defaults to be used.